app/src/main/java/app/gamenative/service/gog/GOGManager.kt

```diff
@@ -0,0 +1,904 @@
+    }
+
+
+    fun verifyInstallation(gameId: String): Pair<Boolean, String?> {
+        val game = runBlocking { getGameById(gameId) }
+        val installPath = game?.installPath
+
+        if (installPath == null || !game.isInstalled) {
+            return Pair(false, "Game not marked as installed in database")
+        }
+
+        val installDir = File(installPath)
+        if (!installDir.exists()) {
+            return Pair(false, "Install directory not found: $installPath")
+        }
+
+        if (!installDir.isDirectory) {
+            return Pair(false, "Install path is not a directory")
+        }
+
+        val contents = installDir.listFiles()
+        if (contents == null || contents.isEmpty()) {
+            return Pair(false, "Install directory is empty")
+        }
+
+        Timber.i("Installation verified for game $gameId at $installPath")
+        return Pair(true, null)
+    }
+
+    // Get the exe. There is a v1 and v2 depending on the age of the game.
+    suspend fun getInstalledExe(context: Context, libraryItem: LibraryItem): String = withContext(Dispatchers.IO) {
+        val gameId = libraryItem.gameId.toString()
+        try {
+            val game = getGameById(gameId) ?: return@withContext ""
+            val installPath = getGameInstallPath(context, game.id, game.title)
+
+            // Try V2 structure first (game_$gameId subdirectory)
+            val v2GameDir = File(installPath, "game_$gameId")
+            if (v2GameDir.exists()) {
+                return@withContext getGameExecutable(installPath, v2GameDir)
+            }
+
+            // Try V1 structure
+            val installDirFile = File(installPath)
+            val subdirs = installDirFile.listFiles()?.filter {
+                it.isDirectory && it.name != "saves"
+            } ?: emptyList()
+
+            if (subdirs.isNotEmpty()) {
+                return@withContext getGameExecutable(installPath, subdirs.first())
+            }
+
+            ""
+        } catch (e: Exception) {
+            Timber.e(e, "Failed to get executable for GOG game $gameId")
+            ""
+        }
+    }
+
+    private fun getGameExecutable(installPath: String, gameDir: File): String {
+        val result = getMainExecutableFromGOGInfo(gameDir, installPath)
+        if (result.isSuccess) {
+            val exe = result.getOrNull() ?: ""
+            Timber.d("Found GOG game executable from info file: $exe")
+            return exe
+        }
+        Timber.e(result.exceptionOrNull(), "Failed to find executable from GOG info file in: ${gameDir.absolutePath}")
+        return ""
+    }
+
+    private fun getMainExecutableFromGOGInfo(gameDir: File, installPath: String): Result<String> {
+        return try {
+            val infoFile = gameDir.listFiles()?.find {
+                it.isFile && it.name.startsWith("goggame-") && it.name.endsWith(".info")
+            } ?: return Result.failure(Exception("GOG info file not found in ${gameDir.absolutePath}"))
```

Comment:
This is dangerous, can lead to crashes if there are many files in the folder. Has happened before for custom games. We should try to look in an efficient way like we do there.

Reply:
Also, does this handle the file being in nested folders? For KOTOR, it is. 
<img width="815" height="135" alt="image" src="https://github.com/user-attachments/assets/d0587fda-e5dc-4a99-9bbf-c1d598dedbb7" />

Reply:
Nvm, I believe listFiles() is efficient?